schematic/store/synapse.py

```diff
@@ -387,96 +462,79 @@ def getFilesInStorageDataset(
 
         return file_list
 
+    def _get_manifest_id(self, manifest: pd.DataFrame) -> str:
+        """If both censored and uncensored manifests are present, return uncensored manifest; if only one manifest is present, return manifest id of that manifest; if more than two manifests are present, return the manifest id of the first one. 
+        Args:
+        manifest: a dataframe contains name and id of manifests in a given asset view
+
+        Return: 
+        manifest_syn_id: id of a given censored or uncensored manifest
+        """ 
+        censored_regex=re.compile('.*censored.*')
+        censored = manifest['name'].str.contains(censored_regex)
+        if any(censored):
+            # Try to use uncensored manifest first
+            not_censored=~censored
+            if any(not_censored):
+                manifest_syn_id=manifest[not_censored]["id"][0]
+            # if only censored manifests are available, just use the first censored manifest
+            else: 
+                manifest_syn_id = manifest["id"][0]
```

Comment:
@linglp this is actually a big assumption here. If someone isn't super familiar with the file structure expected and is just trying things out, this might return some confusing results. It would be nice to check if there are multiple manifests that _could_ be downloaded to count them, and raise a warning telling the user that its expected to only have a single manifest (two if censored and uncensored) and that you are just going to pull a single manifest (providing that manifest id).

Reply:
This applies to lines 482 and 486.

Reply:
Thanks for bringing this up. I think here i tried to preserve the old behavior that we have in our code base (see [here](https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/store/synapse.py#:~:text=manifest_syn_id%3Dmanifest%5B%22id%22%5D%5B0%5D)) . If I understand it correctly, if there are uncensored manifest and censored manifest, the code would first try using the uncensored manifest. If there are multiple uncensored manifest, we would use the first one? And if there are multiple censored manifests available, we would download the first one as well. I agree with you that issuing a warning to the users and explaining the situation is better. Should I address the problem here or open a separate PR to address in develop?

Reply:
Why don't you create an issue in Jira and use is situation as a user story.

Reply:
Thanks Mialy. I wrote a jira issue here: https://sagebionetworks.jira.com/browse/FDS-345